snoop/models.py

```diff
@@ -40,13 +40,13 @@ class EmailCache(models.Model):
     value = models.TextField()
     time = models.DateTimeField(auto_now=True)
 
-class ArchiveListCache(models.Model):
-    sha1 = models.CharField(max_length=50, primary_key=True)
-    value = models.TextField()
-    time = models.DateTimeField(auto_now=True)
-
 class Document(models.Model):
-    container = models.ForeignKey('Document', null=True)
+    container = models.ForeignKey('Document',
+                                  related_name='document_containers',
```

Comment:
Actually, from the point of view of the container, these will be contained documents, not containers. And, to preserve Django model naming conventions, since the default related name is `document_set`, let's use `contained_set`.